proxy/core/src/main/scala/io/cloudstate/proxy/ProtobufGeneratedMessageRegisterFeature.scala

```diff
@@ -8,26 +8,24 @@ import scala.collection.JavaConverters._
 
 @AutomaticFeature
 final class ProtobufGeneratedMessageRegisterFeature extends Feature {
-
-  override def duringAnalysis(access: Feature.DuringAnalysisAccess) =
-    List(
-      "akka.protobuf.GeneratedMessage",
-      "akka.protobuf.GeneratedMessage$Builder",
-      "akka.protobuf.ProtocolMessageEnum",
-      "com.google.protobuf.GeneratedMessageV3",
-      "com.google.protobuf.GeneratedMessageV3$Builder",
-      "com.google.protobuf.ProtocolMessageEnum",
-      "scalapb.GeneratedMessage"
-    ).foreach { className =>
-      val cls = access.findClassByName(className)
-      if (cls != null) reachableSubtypes(access, cls).foreach { cls =>
-        RuntimeReflection.register(cls)
-        RuntimeReflection.register(cls.getMethods: _*)
-      }
+  val messageClasses = Vector(
+    classOf[akka.protobuf.GeneratedMessage],
+    classOf[akka.protobuf.GeneratedMessage.Builder[_]],
+    classOf[akka.protobuf.ProtocolMessageEnum],
+    classOf[com.google.protobuf.GeneratedMessageV3],
+    classOf[com.google.protobuf.GeneratedMessageV3.Builder[_]],
+    classOf[com.google.protobuf.ProtocolMessageEnum],
+    classOf[scalapb.GeneratedMessage]
+  )
+  override final def duringAnalysis(access: Feature.DuringAnalysisAccess) =
+    messageClasses.foreach { cls =>
+      if (access.isReachable(cls)) {
+        access.reachableSubtypes(cls).iterator.asScala.filter(_ != null) foreach {
+          case null => // Java…
```

Comment:
lol, wat?

Reply:
LOL—well, the PF-literal predated the filter, I'll remove the PF literal and replace it with a function.

Reply:
Phew.  I thought for some "Java..." reasons `.filter(_ != null) foreach { case null => ` as actually being executed _somehow_!

Reply:
That would have been extremely unnerving